File: vs/compile.js
#!/usr/bin/env node

var exec = require('child_process').exec;

function compile (expr) {
  console.log( "Preparing...");

  var cmd = 'rm -rf classes && mkdir classes && ';

  if(expr == undefined || expr == null)
    cmd += "scalac -d classes src/*.scala";
  else if (expr.indexOf(".scala") > -1)
    cmd += "scalac -d classes " + expr ;
  else
    cmd += "scalac -d classes " + expr + ".scala" 

  cmd += " -feature" 
 
  console.log( "Compiling: " + cmd);
  exec( cmd, 
    function (error, stdout, stderr) {
      if(error)
        console.log('ERROR: ' + error)
      else 
        if( stdout)
          console.log(stdout);
      console.log( "Done.");
    });
}

if( process.argv.length < 3 ) {
  console.log( "No File specified. Using *");
  compile();
} else
  compile( process.argv[2] );

File: .snippets/github.com/khilnani/snippets/vs/compile.rb
#!/usr/bin/ruby

def compile (expr="*")
  puts "Preparing ..." 

  system "rm -rf classes"
  system "mkdir classes"

  cmd = String.new("")

  if expr == "*"
    cmd = "scalac -d classes src/*.scala"
  elsif expr.end_with? ".scala"
    cmd = "scalac -d classes #{expr}"
  else
    cmd = "scalac -d classes #{expr}.scala" 
  end

  cmd += " -feature"

  puts "Compiling: #{cmd}"
  system cmd

  puts "Done."
end

if ARGV.length == 0
  puts "No File specified. Using *"
  compile
else
  compile ARGV[0]
end
